pkg/connector/matrixfmt/convert.go

```diff
@@ -0,0 +1,18 @@
+package matrixfmt
+
+import (
+	"context"
+
+	"maunium.net/go/mautrix/event"
+)
+
+func Parse(ctx context.Context, parser *HTMLParser, content *event.MessageEventContent) string {
+	if content.FormattedBody == "" {
+		return content.Body
+	}
+
+	parseCtx := NewContext(ctx)
+	parseCtx.AllowedMentions = content.Mentions
+	parsed := parser.Parse(content.FormattedBody, parseCtx)
+	return string(parsed.String)
```

Comment:
Looks like this drops the parsed entities? Are they actually needed? If not, there's no point in having all the EntityString stuff, just the standard HTMLParser from mautrix-go is sufficient with a customized mention converter